Svc/RateGroupDriver/RateGroupDriver.cpp

```diff
@@ -12,29 +12,30 @@ namespace Svc {
 
     }
 
-    void RateGroupDriver::configure(NATIVE_INT_TYPE dividers[], NATIVE_INT_TYPE numDividers)
+    void RateGroupDriver::configure(DividerSet dividersSet)
     {
 
         // check arguments
-        FW_ASSERT(dividers);
-        FW_ASSERT(numDividers);
-        this->m_numDividers = numDividers;
-        FW_ASSERT(numDividers <= static_cast<NATIVE_INT_TYPE>(FW_NUM_ARRAY_ELEMENTS(this->m_dividers)),
-                numDividers,
+        FW_ASSERT(dividersSet.dividers);
+        this->m_numDividers = RateGroupDriver::DIVIDER_SIZE;
+        FW_ASSERT(this->m_numDividers <= static_cast<NATIVE_INT_TYPE>(FW_NUM_ARRAY_ELEMENTS(this->m_dividers)),
+                this->m_numDividers,
                 static_cast<NATIVE_INT_TYPE>(FW_NUM_ARRAY_ELEMENTS(this->m_dividers)));
         // verify port/table size matches
         FW_ASSERT(FW_NUM_ARRAY_ELEMENTS(this->m_dividers) == this->getNum_CycleOut_OutputPorts(),
                 static_cast<NATIVE_INT_TYPE>(FW_NUM_ARRAY_ELEMENTS(this->m_dividers)),
                 this->getNum_CycleOut_OutputPorts());
-        // clear table
-        ::memset(this->m_dividers,0,sizeof(this->m_dividers));
         // copy provided array of dividers
-        for (NATIVE_INT_TYPE entry = 0; entry < numDividers; entry++) {
-            this->m_dividers[entry] = dividers[entry];
+        for (NATIVE_UINT_TYPE entry = 0; entry < RateGroupDriver::DIVIDER_SIZE; entry++) {
+            // A port with an offset equal or bigger than the divisor is not accepted because it would never be called
+            FW_ASSERT((this->m_dividers[entry].offset==0)||(this->m_dividers[entry].offset < this->m_dividers[entry].divisor),
```

Comment:
Should this not assert `dividersSet.dividers`, which is the passed-in argument, rather than `this->m_dividers` which is initialized to 0 and is going to be overwritten down below anyways?

Reply:
Yes.